core/db/sql/writes.sql

```diff
@@ -15,3 +15,27 @@ values ($1, $2);
 -- name: InsertRegisteredNode :exec
 insert into core_validators(pub_key, endpoint, eth_address, comet_address, eth_block, node_type, sp_id)
 values ($1, $2, $3, $4, $5, $6, $7);
+
+-- name: UpsertSlaRollupReport :exec
+with updated as (
+    update sla_node_reports 
+    set blocks_proposed = blocks_proposed + 1
+    where address = $1 and sla_rollup_id is null
+    returning *
+)
+insert into sla_node_reports (address, blocks_proposed, sla_rollup_id)
+select $1, 1, null
+where not exists (select 1 from updated);
+
+-- name: ClearUnindexedSlaNodeReports :exec
+delete from sla_node_reports
+where sla_rollup_id is null;
+
+-- name: IndexSlaNodeReport :exec
+insert into sla_node_reports (sla_rollup_id, address, blocks_proposed)
+values ($1, $2, $3);
+
+-- name: IndexSlaRollup :one
+insert into sla_rollups (time, block_start, block_end)
+values ($1, $2, $3)
+returning id;
```

Comment:
To keep the db verbiage intact let's use something like `Save` or `Insert` instead of the prefix `Index`. That way when we go to implement the data companion (indexer) the generated code won't conflict naming wise with that.